src/xmipp/bindings/python/xmipp_base.py

```diff
@@ -98,7 +110,321 @@ def tryRun(self):
             import traceback
             traceback.print_exc(file=sys.stderr)
             return 1
-            
+
+    @classmethod
+    def getModel(cls, *modelPath, **kwargs):
+        """ Returns the path to the models folder followed by
+            the given relative path.
+        .../xmipp/models/myModel/myFile.h5 <= getModel('myModel', 'myFile.h5')
+
+            NOTE: it raise and exception when model not found, set doRaise=False
+                  in the arguments to skip that raise, especially in validation
+                  asserions!
+        """
+        return getModel(*modelPath, **kwargs)
+
+    @classmethod
+    def runCondaCmd(cls, program, arguments, **kwargs):
+        """ This class method is used to run programs that are independent of xmipp
+            but employ conda. The class should possess a _conda_env attribute to be used.
+            Otherwise  CONDA_DEFAULT_ENVIRON is used. To use xmipp dependent
+            programs, runCondaJob within a XmippProtocol is preferred.
+                :param program:str. A program/pythonScript to execute
+                :param arguments: str. The arguments for the program
+                :param kwargs: options
+                :return: None
+        """
+        if "env" not in kwargs:
+            kwargs["env"] = os.environ
+
+        if hasattr(cls, "_conda_env"):
+            condaEnvName = cls._conda_env
+        else:
+            condaEnvName = CondaEnvManager.CONDA_DEFAULT_ENVIRON
+            print("Warning: '_conda_env' not defined in '%s'. "
+                  "Using the default one ('%s')" % (cls, condaEnvName))
+
+        program, arguments, kwargs = prepareRunConda(program, arguments,
+                                                     condaEnvName, **kwargs)
+
+        cmd_args = program + " " + arguments
+        print(cmd_args)
+        try:
+            subprocess.check_call(cmd_args, shell=True, **kwargs)
+        except subprocess.CalledProcessError as e:
+            subprocess.check_call(cmd_args, shell=True, **kwargs)
+
+
+def prepareRunConda(program, arguments, condaEnvName, **kwargs):
+    '''
+    Used to get the arguments for the methods runCondaJob or runCondaCmd
+    :param program: string
+    :param arguments: string
+    :param condaEnvName: string
+    :param kwargs:
+    :return: (program, arguments, kwargs). Updated values
+    '''
+    # print(kwargs['env'])
+    kwargs['env'] = CondaEnvManager.modifyEnvToUseConda(kwargs['env'],
+                                                        condaEnvName)
+
+    kwargs['env']['PYTHONWARNINGS'] = 'ignore::FutureWarning'
+    usePython = False
+    programFn = os.path.join(getXmippPath('bin'), program)
+
+    try:
+        with open(programFn) as f:
+            if "python" in f.read(32):
```

Comment:
this will cause some troubles

Reply:
Which part exactly. The if "python" in f.read(32)?

Reply:
Yes. Imagine somebody writing C++ program like 'convertWithPython' ...

Reply:
I agree. This was more a dirty solution than a final one. Any Idea of how to detect if something is a python script, appart from trying to execute it?

Reply:
you removed the extension, right?

Reply:
if you control all the python scripts, just add the same first line and test for the whole line?